### PR TITLE
Docs - `plugin register` added to internals with difference in registering for prior versions noted

### PR DESCRIPTION
**Pull request by aphorise**

### CODE REVIEW

File: website/pages/docs/internals/plugins.mdx [hunk 1]
```diff
@@ -88,12 +88,24 @@ it instead becomes visible to backends and can be executed by them. For more
 information on the plugin catalog please see the [Plugin Catalog API
 docs](/api/system/plugins-catalog).
 
-An example plugin submission looks like:
+In current versions of Vault - plugin submission resemble:
```
Comment (kalafut) [inline website/pages/docs/internals/plugins.mdx:91]:
```suggestion
An example of plugin registration in current versions of Vault:
```

File: website/pages/docs/internals/plugins.mdx [hunk 2]
```diff
@@ -88,12 +88,24 @@ it instead becomes visible to backends and can be executed by them. For more
...
+```shell-session
+$ vault plugin register -sha256=<SHA256 Hex value of the plugin binary> \
+    secret \                  # type
+    myplugin-database-plugin
+
+Success! Registered plugin: myplugin-database-plugin
+```
+
+Vault versions prior to v0.10.4 lacked the `vault plugin` operator and the 
+registeration step for them are:
```
Comment (kalafut) [inline website/pages/docs/internals/plugins.mdx:102]:
```suggestion
registration step for them is:
```

**Comment by aphorise** [comment]
@kalafut - do you know if this will be merged anytime soon? - or perhaps I'm still missing something?

**Comment by aphorise** [comment]
Can this be merged? - or is there anything that may be missing?

**Comment by raskchanky** [comment]
@aphorise It looks like @kalafut made some suggestions which you resolved without implementing. I would suggest making those changes.

**Comment by kalafut** [comment]
The content looks fine, but since the PR we've rearranged the docs file layout a bit. Can you please resolve that merge conflict?

**Comment by aphorise** [comment]
Redone PR on #10864.

**Pull request closed without merging**